src/DotNetProjectFile.Analyzers/Analyzers/MsBuild/PlaceXmlNodesOnSeperateLines.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System.IO;
+using System.Xml;
+
+namespace DotNetProjectFile.Analyzers.MsBuild;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+public sealed class PlaceXmlNodesOnSeperateLines() : MsBuildProjectFileAnalyzer(Rule.PlaceXmlNodesOnSeperateLines)
+{
+    protected override void Register(ProjectFileAnalysisContext context)
+    {
+        var nodes = GetFlatChildren(context.Project).ToArray();
+        var occupied = new Dictionary<int, Node>();
+
+        foreach (var node in nodes)
+        {
+            (var startLine, var endLine) = GetLineNumbers(node);
+
+            if (occupied.TryGetValue(startLine, out var prev) && prev != node)
+            {
+                context.ReportDiagnostic(Descriptor, node, node.ToString());
+                continue;
+            }
+            else
+            {
+                occupied[startLine] = node;
+            }
+
+            if (occupied.TryGetValue(endLine, out prev) && prev != node)
+            {
+                context.ReportDiagnostic(Descriptor, node, node.ToString());
+            }
+            else
+            {
+                occupied[endLine] = node;
+            }
+        }
+    }
+
+    private static IEnumerable<Node> GetFlatChildren(Node node)
+    {
+        foreach (var child in node.Children)
+        {
+            yield return child;
+            foreach (var grandchild in GetFlatChildren(child))
+            {
+                yield return grandchild;
+            }
+        }
+    }
+
+#pragma warning disable S3776 // Max complexity of 15. Justification: Logic state machine relatively well contained.
+    private static (int StartLine, int EndLine) GetLineNumbers(Node node)
```

Comment:
Why is this needed? node.LineInfo.LinePositionSpan() gives you this already. And depth can be resolved with `node. AncestorsAndSelf().Count()`;

Reply:
It only gives you the starting line, not the ending line

Reply:
It gives both, but the end line is wrong. I'd like to fix that differently. See #136

Reply:
I would be in favour of closing this one. #135 also covers this one.

Reply:
I would like to give the changes in #135 a shot at my test case and adjust the expected results accordingly. If this indeed produces equivalent warnings, I would still like to explicitly add the test cases to the test suite (while removing all other logic currently present).

Reply:
Seems to work indeed:

```
Verification failed: 4 extra, 4 missing.
[+] Proj1700 The element <TargetFramework> has not been properly indented. @[(2,17)-(2,33)]
[+] Proj1700 The element </ItemGroup> has not been properly indented. @[(6,74)-(6,86)]
[+] Proj1700 The element <PackageReference> has not been properly indented. @[(4,13)-(4,101)]
[+] Proj1700 The element </ItemGroup> has not been properly indented. @[(9,43)-(9,55)]
[-] Proj1700 Move XML node '<TargetFramework>net8.0</TargetFramework>' to a new line. @[(2,18)-(2,19)]
[-] Proj1700 Move XML node '<PackageReference Include="Microsoft.CodeAnalysis.Workspaces.Common" Version="4.10.0" />' to a new line. @[(4,14)-(4,101)]
[-] Proj1700 Move XML node '<PackageReference Include="Qowaiv.Analyzers.CSharp" Version="2.0.0" />' to a new line. @[(6,5)-(6,6)]
[-] Proj1700 Move XML node '<Compile Include="../common/Code.cs" />' to a new line. @[(9,5)-(9,6)]
```